tests/test_extract.py

```diff
@@ -9,12 +12,51 @@
 from . import I18nToolTestCase, MOCK_DJANGO_APP_DIR
 
 
+def extract_options():
```

Comment:
I can split it into three commits, would that help @OmarIthawi ?

* feat: move babel work from `Extract.run` into a method to avoid Long Methon linter warnings
* feat: refactor `extract` tests to make them more flexible
* feat: add two new options, merge_js and overrite_from_partial

Reply:
Thank you so much!!